query/src/main/java/org/infinispan/query/dsl/embedded/impl/LuceneQueryMaker.java

```diff
@@ -115,6 +116,28 @@ public LuceneQueryMaker(SearchIntegrator searchFactory, FieldBridgeProvider<Type
          isAnalyzerRemote = true;
       }
       Query query = makeQuery(parsingResult.getWhereClause());
+
+      // an all negative top level boolean query is not allowed; needs a bit of rewriting
+      if (query instanceof BooleanQuery) {
+         BooleanQuery booleanQuery = (BooleanQuery) query;
+         boolean allClausesAreMustNot = true;
+         for (BooleanClause clause : booleanQuery.clauses()) {
+            if (clause.getOccur() != BooleanClause.Occur.MUST_NOT) {
+               allClausesAreMustNot = false;
+               break;
+            }
+         }
+         if (allClausesAreMustNot) {
+            //It is illegal to have only must-not queries, in this case we need to add a positive clause to match everything else.
```

Comment:
Because Lucene is weird, and "-text:foo" never matches anything.
But "-text:foo #*:*" does match what we intend.